txscript/engine.go

```diff
@@ -1033,6 +1060,22 @@ func (vm *Engine) Execute() (err error) {
 		return nil
 	}
 
+	// If the StepCallback is set, we start by making a call back with the
+	// initial engine state.
+	var stepInfo *StepInfo
+	if vm.StepCallback != nil {
+		stepInfo = &StepInfo{
+			ScriptIndex: vm.scriptIdx,
+			OpcodeIndex: vm.opcodeIdx,
+			Stack:       vm.dstack.stk,
```

Comment:
Perhaps this should actually be a _copy_ so the callback doesn't inadvertently affect the script execution (mutation, etc). Main application here seems to be debugging or introspection, so doesn't seem to be as sensitive to extra allocations.

Reply:
Also for your use case, would it be easier to obtain the string version (like called below), or just leave it with the stack?

Reply:
Fixed: https://github.com/btcsuite/btcd/pull/1980/commits/55a376e6ea591ba5a6e3ed2302787c95c8135d06

It is useful to be able to inspect the raw stack instead of the string version.